File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.java
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> dfs = new ArrayList<>();
        if(root == null) return dfs;
        Queue<TreeNode> q = new LinkedList<>();
        q.add(root);

        while(q.size()>0) {
            int n = q.size();
            List<Integer> sub = new ArrayList<>();
            while(n>0) {
                TreeNode node = q.remove();
                if(node.left != null) q.add(node.left);
                if(node.right != null) q.add(node.right);
                sub.add(node.val);
                n--;
            }
            dfs.add(sub);
        }
        return dfs;
    }
}

File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.cpp
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ans;
        if (!root) return ans; // ✅ handle empty tree case

        queue<TreeNode*> q;
        q.push(root);

        while (!q.empty()) {
            vector<int> x;
            int n = q.size();

            for (int i = 0; i < n; i++) {
                TreeNode* temp = q.front();
                q.pop();
                x.push_back(temp->val);
                if (temp->left) q.push(temp->left);
                if (temp->right) q.push(temp->right);
            }

            ans.push_back(x);
        }

        return ans;
    }
};
